### ISSUE TITLE
A script that remove all deprecated preferences in Prefs.js

### ISSUE THREAD

**Issue by PureOcean** [arkenfox/user.js#1720]
It's known that some preferences in `Prefs.js` file are deprecated in each new Firefox version. Unless the profile is completely reset, they keep to exist and increase the file size of `Pref.js`. Yes, deprecated preferences can appear as a trash icon in `about:config`, but it's not possible to delete them all at once.

[Arkenfox-cleanup.js](https://github.com/arkenfox/user.js/blob/master/scratchpad-scripts/arkenfox-cleanup.js) clears only deprecated preferences from its User.js list. My goal is to clean up all the deprecated preferences which indicated by the trash icon, in Firefox's own listed, even if there is a risk that some outdated extensions need deprecated preferences.

I hope you JavaScript experts can prepare a Script that can achieve this. The source code of about:config might be useful: `chrome://global/content/aboutconfig/aboutconfig.js`

**Comment by rusty-snake** [comment]
> deprecated preferences can appear as a trash 

FTR: Hidden prefs also have a trash icon.

**Comment by rusty-snake** [comment]
> I hope you JavaScript experts can prepare a Script that can achieve this. 

So anyhow related to arkenfox beyond 'Firefox prefs'?

**Comment by Thorin-Oakenpants** [comment]
this is out of scope for this project. We have a cleanup script for prefs we have mentioned/flipped in the past, and I keep deprecated and removed prefs in the user.js for each full ESR cycle - anything outside of that is not my problem

**Comment by PureOcean** [comment]
@rusty-snake and @Thorin-Oakenpants,
I am grateful for your replies. Also your efforts to improve the Arkenfox user.js and cleaner+Updater BAT files.

**Comment by Thorin-Oakenpants** [comment]
deprecated prefs won't cause any issues, but if you're neuro divergent about them (and I've checked mine from time to time in about:config - just check the `show only modified preferences`) then you have two options
- use a new profile
- collate all the deprecated prefs from all the diff issues in this repo (I think we go back to FF52), and add to the current cleanup script and run

**Comment by PureOcean** [comment]
@Thorin-Oakenpants,
 
Thank you for your suggestions. I did research before reading your suggestions.

> resource://gre/modules/Preferences.sys.mjs
> https://searchfox.org/mozilla-central/source/toolkit/modules/Preferences.sys.mjs
> https://searchfox.org/mozilla-central/source/browser/app/profile/firefox.js
> https://searchfox.org/mozilla-central/source/modules/libpref/init/all.js
> 

These resources could have been useful for prepare script but most probably script's result would be harmful. Because all `extensions.*` prefs are also removable except Hidden. 

You're right. The healthiest solution is create a new profile. Of course, need backup the personal prefs to the `user-override.js` file. For now, I just tried your second suggestion. Great idea. I appreciate you. I also would like to take this occasion to thank @earthlng for [Diffs lists](https://github.com/arkenfox/user.js/issues?q=ToDo%3A+diffs+label%3Adiffs).
